test/system/252-quadlet.bats

```diff
@@ -127,6 +127,8 @@ function service_cleanup() {
                "state of service $service after systemctl stop"
     fi
 
+    # reset-failed necessary to clean up stray systemd cruft
+    run systemctl reset-failed "$service"
```

Comment:
Doesn't this also/only need to happen in teardown? Because if the test fails in the middle we never get to the point where service_cleanup is called. Basically the same thing you did in 250-systemd.bats

Reply:
Yes, so done. Thank you.